File: algorithms/14. Longest Common Prefix/solution.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        lcp = strs[0] if strs else ""
        while True:
            for s in strs:
                if s.find(lcp) != 0:
                    lcp = lcp[:-1]
                    break
            else:
                break
        return lcp

File: algorithms/14. Longest Common Prefix/solution.cc
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int n = strs.size(), pos = 0;
        string lcp;
        if ( n < 1 ) {
            return lcp;
        }
        while ( true ) {
            for ( int i = 0; i < n; ++i ) {
                if ( pos >= strs[i].size() ) {
                    return lcp;
                }
                if ( i > 0 && strs[i][pos] != strs[i-1][pos] ) {
                    return lcp;
                }
            }
            lcp += strs[0][pos];
            ++pos;
        }
    }
};
